batcher/batcher_test.go

```diff
@@ -0,0 +1,105 @@
+package batcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb"
+)
+
+// TestBatch_Size ensures that a batcher generates a batch when the size threshold is reached.
+func TestBatch_Size(t *testing.T) {
+	batchSize := 5
+	batcher := NewBatcher(batchSize, time.Hour)
+	if batcher == nil {
+		t.Fatal("failed to create batcher for size test")
+	}
+
+	in := make(chan influxdb.Point)
+	out := make(chan []influxdb.Point)
+
+	go batcher.Start(in, out)
+
+	var p influxdb.Point
+	for i := 0; i < batchSize; i++ {
```

Comment:
In the case where you expect 5 points, 3 points could of actually sent out, but you'll never catch that in this test because you are forcing the test to work synchronously.  I'm not sure why we don't want to test that functionality.  Unless it's being tested later and I missed it.